content/post/gld.md

```diff
@@ -0,0 +1,276 @@
+
+Normalized-2 - dots and copyrights removed:
+{{% code src="/post/gld/norm/5.txt" lang="nohighlight" height="400" %}}
+
+Normalized-3 - non-alphanumeric symbols removed:
+{{% code src="/post/gld/norm/6.txt" lang="nohighlight" height="400" %}}
+
+We use level 3 output for LSH and level 2 output for the Levenshtein distance refinement.
+
+## Merged licenses
+
+There are projects with several license texts in the same file
+Some of them are dual-licensed, some mention dependencies. Our core matcher assumes the single
+sample by design, and those cases are hard for it to handle.
+[google/licenseclassifier](https://github.com/google/licenseclassifier) project gracefully
+digests them because it is based on ngram hashing instead, and hence we considered switching to that
+algorithm. We did not for the following reasons:
+
+1. The memory consumption is higher, about [200 MB](https://github.com/google/licenseclassifier/blob/master/classifier.go#L188).
+2. Non-trivial database preprocessing and lack of high-level documentation.
+3. Slower performance on single licenses.
+3. It appeared that 95% of the cases could be resolved by simple split heuristics.
+
+After all, we found that since it is very cheap to query a text, we could make a few split assumptions
+and process each variant independently. For example, many texts separate licenses
+with `===` or `---` decorations. Besides, it is common to place the license name before the body
+and we easily find all possible split points. This is still not bulletproff as in license-classifier,
+but as was written works reasonably well.
+
+## Pointers
+
+Sometimes, all the efforts fail and we do not discover anything in `LICENSE` files and in `README` files.
+There is still hope: we discovered that many projects contain the URL to the official license text
+in them. E.g. [awesome lists which have the CC0 badge at the bottom](https://github.com/terryum/awesome-deep-learning-papers),
+[Apache banners](https://github.com/dmlc/xgboost/blob/master/LICENSE) or numerous users of
+[mit-license.org](https://mit-license.org). 
+
+## Implementation
+
+go-license-detector is a library and a self-contained binary CLI tool.
+
+[Francesc](https://twitter.com/francesc), our VP of Developer Community, took a serious effort
+in making the code idiomatic. You may have experienced this: when you are focused on **what** your
+code is doing, you often miss **how** the code is looking. I would like to write
+a follow-up post which describes which points were improved and what were the
+typical issues.
+
+## Offtopic
+
+Since we had to manually look through hundreds of most-starred projects on GitHub, we noticed
+a few funny trends. Many Chinese repositories isolated from the other communities,
+awesome list expansion and others. Again, I should devote a separate post to those,
+they are funny and also help to understand the picture of open source popularity better.
```

Comment:
I would keep the offtopic section out since you don't really explain what is going on. Better to be a separate post for another day.

Reply:
Rephrased and left the section because I doubt I will find time in the nearest future.